src/crimson/osd/main.cc

```diff
@@ -219,10 +217,9 @@ int main(int argc, char* argv[])
      "Prometheus metrics prefix");
 
   auto [ceph_args, app_args] = partition_args(app, argv, argv + argc);
-  if (ceph_argparse_need_usage(ceph_args) &&
-      std::find(app_args.begin(), app_args.end(), "--help") == app_args.end()) {
+  if (ceph_argparse_need_usage(ceph_args) ||
+      ceph_argparse_need_usage(app_args)) {
```

Comment:
if my proposal above makes sense, we can dispense with the `ceph_argparse_need_usage(app_args)` condition.

Reply:
See above, I added `ceph_argparse_need_usage()` to check both ceph_args and app_args is to make it independent about how `partition_args()` works. IIUC this is more robust to find out `--help` from user input.

Reply:
i see. you don't want to rely on the behavior of `partition_args()` to tell which part of options to check for the `help,h` option.

sounds good to me.